backend/routes/admin.py

```
@@ -348,14 +386,51 @@ def undo_check_in_rsvp(
         "checked_in_at": None,
+        "ride_leader_summary": serialize_ride_leader_snapshot(snapshot),
+    }
+
+
+# ── Admin Ride Leader Actions ────────────────────────────────
+
+@router.post("/api/admin/events/{event_id}/rsvp/ride-leader")
+def activate_ride_leader(
+    event_id: int,
+    body: RideLeaderRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)